Hivelord Designer Improvements.

Venuska1117 opened the pull request:
# Spotted issues or want to give feedback?
click this link: https://discord.com/channels/150315577943130112/1464087612796174501

# About the pull request

Re-organizes file contents to designated files, changes structure of designer.dm for easier search.
Fixes xeno typos from "you" and "me" to "we" and "our" (xeno speech)

Fixes issues introduced with: #7703

Documentation of designer changes: https://hackmd.io/@YLxQa9NUQ-e6H5YHujbzoA/HJVzstldZg

# Explain why it's good for the game

I think i went a little too hard on designer when they were created at first, their general gamplay of not be able making structures would be fine if some functions worked as intend, weedbound sabotaging itself and generaly making it hard-to-play with medium reward price, i think this changes should improve its gameplay to be less painfull to exist and somewhat reward gameplay and coop interactions.


# Testing Photographs and Procedure

<details>
<summary>>Nodes Marks can now be swapped</summary>
No more  need to destroy nodes to correct mark!

https://github.com/user-attachments/assets/01d5b84e-2e4d-4f7a-b760-97d25e70d0db
</details>

<details>
<summary>>New Nodebound variant in action</summary>
This new wall now saves weedbound from sabotaging itself!

https://github.com/user-attachments/assets/23313972-8616-445f-9e8a-19fa64de5c68
</details>

<details>
<summary>>Nodebound morphing</summary>
Nodebound now changes to Weedbound when nodes change!

https://github.com/user-attachments/assets/e7dba716-3033-4665-951d-1f01a4ced4c1

</details>

<details>
<summary>>"Bound" Resin Adaptation</summary>
Now "bound" resin adapt to weeds strenght!

https://github.com/user-attachments/assets/828ea42a-a448-4161-a937-f64032a3cfed

</details>

<details>
<summary>Cursor option for designer:</summary>
Preferences -> UI -> Toggle Custom Cursors.

https://github.com/user-attachments/assets/8c74a819-b871-4cee-a670-812d5918605b

</details>

# Details for Reviewer.

Before i will write notes, you should **review code on commits** that i made instead of inspecting **files changed**, 65bc3d3 push re-organized file and spread some content to selective files, making it look like XXL change, but in reality its around M-L changes. 
> its very messy to look at code thats why i offer checking commits

**Ignore** **a28d27c** changes, i accidently commited it, and later made commit with same contents in **2930834**, you can inspect both commits side-by-side and they have same changes in them. (**2930834** have more changes than **a28d27c**)

</details>


# Changelog

:cl: Venuska1117
balance: Improved Designer strain.
balance: New weedbound variant called "Nodebound" this variant of weedbound only appear when construct node is "stimulated" on top of weed node, destroying this variant will destroy weeds its attached to (same in reverse), this variant doesn't drop sticky resin.
balance: If ranged builder build on any design node, their build delay after construction is set to 0, meaning cooperating with designer as whisperer or queen is more fluid.
balance: Gardener plasma fruits no longer create thick weedbound variant, now when gardener interacts with node it will create thicker weedbound variant.
balance: Weedbound now adapts to weeds its attached to, if weeds are upgraded, weedbound also upgrades. (upgrades on hardened weeds and hive weeds)
balance: thin sticky nutrient strength increased by +1
balance: Remove Node plasma cost decreased from 25 to 0 
balance: Flexible (cost) node cost changed from 60 to 50
balance: Construct node cost changed from 70 to 50, plasma donation is unchanged (70)
add: Custom Cursor for abilities, designer now uses custom cursor for "Influence" choices, this can be enabled by going Preferences -> UI -> Toggle Custom Cursors.
imagetweak: Changed weedbound icons to fit their theme.
fix: Blue design nodes now will check for decreased plasma cost before trying to use plasma.
fix: Minimum and Maximum duration of tackle knockdown is now working as intended.
del: Removed broken modifier that added +30% speed bonus.
/:cl:

Commit:
Basic organization changes.

Commit:
why

Commit:
second push

Commit:
Merge branch 'designer-improvements' of https://github.com/Venuska1117/cmss13 into designer-improvements

Detective-Google commented:
At the moment I don't believe design needs to be strengthened; it already performs reasonably well.

I also do not believe that designer needs more complexity-- it is already at a complexity level that makes it difficult to understand what it is doing both as an ally and an enemy (overdesigned; one could say.)

In light of this, I don't believe this PR should go forward as it currently is, though any other maintainer is free to remove this DNM if they believe otherwise.

Commit:
detective-google (request?)

Detective-Google commented:
DNM removed as the speed armor and health buffs have been removed.(some technically fixes; but buffs in practice.)

Detective-Google commented on code/__DEFINES/xeno.dm:
```diff
@@ -346,6 +346,7 @@
 #define XENO_SPEED_TIER_2 0.2
 #define XENO_SPEED_TIER_3 0
 #define XENO_SPEED_TIER_4 -0.2
+#define XENO_SPEED_DESIGNER -0.35
```
no longer needed

Commit:
forgor

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Commit:
Adds mouse pointers for designer abilities.

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

Venuska1117 committed:
Merge branch 'master' into designer-improvements

Commit:
small fixes:
- Fixed nodebound icons.
- Added new cursor preference options.
- Designer cursor dont display without preference active.

Venuska1117 committed:
Merge branch 'cmss13-devs:master' into designer-improvements

Commit:
strong weeds upgrade "bound"

Commit:
visual upgrades and code cleaning.
- Added more variants for mouse icons.
- Removed plasma fruit node feeding.
- Added new visuals for node/weedbound and their building effect.

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

Commit:
spell

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

Commit:
Blue nodes Suggestion (fix)
- Friedel Krafts | BRA4, requested blue nodes decreased cost to be checked first before check of structure, because you could have plasma to build, but initial check made it impossible, now if you have enough plasma for cheap build it will work.

Commit:
description fix

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Commit:
fix after conflict

Commit:
plasma cost changes:
- Remove node cost changed from 25 to 0
- Flexible (cost) node cost changed from 60 to 50
- Construct node cost changed from 70 to 50, plasma donation is unchanged (70)

Drulikar commented:
Fix your changelog. Its too ambiguous.

Venuska1117 commented:
> Fix your changelog. Its too ambiguous.

Done, it should be everything i remembered in cl's changes, if i missed something please call out.

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Commit:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

Commit:
helper text

Drulikar commented:
Stale exempt until reviewed

fira commented on code/game/turfs/walls/wall_types.dm:
```diff
@@ -928,8 +929,9 @@
-		qdel(weed_node)
+	if(!destroy_override)
+		for(var/obj/effect/alien/weeds/node/weed_node in contents)
+			qdel(weed_node)
```
You should avoid qdeling stuff in Initialize during mapload, it messes with refs in SSatoms

fira commented on code/modules/mob/living/carbon/xenomorph/strains/castes/hivelord/designer.dm:
```diff
@@ -63,1193 +61,1334 @@
-	layer = FIREDOOR_CLOSED_LAYER
+//////////////////////////
+///   Change Design    ///
+//////////////////////////
```
Don't do that, it gets treated as code documentation of the next definition

Use something else like /* =========

fira commented on code/modules/mob/living/carbon/xenomorph/strains/castes/hivelord/designer.dm:
```diff
@@ -63,1193 +61,1334 @@
+/datum/action/xeno_action/activable/place_design/proc/can_remote_build()
+	if(!locate(/obj/effect/alien/weeds) in get_turf(owner))
+		return FALSE
+	return TRUE
```
shouldn't that check for hive? just wondering, i'm clueless about xeno code

Venuska1117 replied:
this checks if owner (player) is standing on weeds or not, its mainly used by /datum/action/xeno_action/activable/place_design/use_ability(), there is check in place_design for hive ownership

Venuska1117 replied:
but yes, there is no check if weeds under us are ours or not, so technically you could stand on other hive weeds and influence your own weeds (you cannot influence enemy one)

fira commented on code/modules/mob/living/carbon/xenomorph/strains/castes/hivelord/designer.dm:
```diff
@@ -63,1193 +61,1334 @@
-		return FALSE
+///////////////////////////////
+///   Change Node Marker    ///
+///////////////////////////////
```
not doc

fira commented on code/modules/mob/living/carbon/xenomorph/strains/castes/hivelord/designer.dm:
```diff
@@ -63,1193 +61,1334 @@
-		return
+//--------------------------------//
+//-----// Design Node Base //-----//
+//--------------------------------//
```
That's ok 👍

fira commented on code/modules/mob/living/carbon/xenomorph/strains/castes/hivelord/designer.dm:
```diff
@@ -63,1193 +61,1334 @@
-	damage_cap = HEALTH_WALL_XENO_THICK
-	walltype = WALL_THICK_WEEDBOUND_RESIN
+	if(bound_weed.hivenumber != bound_xeno.hivenumber)
+		qdel(src)
```
use INITIALIZE_HINT_QDEL like above

Venuska1117 committed:
i hope its whats needed

Venuska1117 committed:
Merge branch 'master' of https://github.com/Venuska1117/cmss13 into designer-improvements

Venuska1117 committed:
lets hope this decreases size

Venuska1117 committed:
small fixes

fira commented on code/modules/cm_aliens/XenoStructures.dm:
```diff
@@ -1469,3 +1469,732 @@
+		visible_message(SPAN_XENOWARNING("The node withers away."))
+		qdel(src)
+
+/obj/effect/alien/resin/design/proc/unregister_weed_expiration_signal_design()
```
no big deal but that doesn't really need to be a separate proc

Venuska1117 replied:
changing that causes CI checks to fail, it is used by weeds.dm to get signal for bound_weeds owner, changing that will cause failure

fira commented on code/modules/cm_aliens/XenoStructures.dm:
```diff
@@ -1469,3 +1469,732 @@
+			else
+				placed = target_turf.place_on_top(/turf/closed/wall/resin/weedbound/normal)
+
+			var/turf/closed/wall/resin/Res = get_turf(target_turf)
```
```suggestion
			var/turf/closed/wall/resin/Res = target_turf
```

fira commented on code/modules/cm_aliens/XenoStructures.dm:
```diff
@@ -1469,3 +1469,732 @@
+	if(!bound_weed)
+		addtimer(CALLBACK(src, PROC_REF(check_weed_replacement)), 3 SECONDS)
+		return
+	if(bound_weed)
```
if redundant here

fira commented on code/modules/cm_aliens/XenoStructures.dm:
```diff
@@ -1469,3 +1469,732 @@
+		UnregisterSignal(bound_weed, COMSIG_PARENT_QDELETING)
+		bound_weed = null
+		for(var/obj/effect/alien/weeds/node/weed_node in contents)
+			qdel(weed_node)
```
Why not just qdel bound_weed?

fira commented on code/modules/cm_aliens/XenoStructures.dm:
```diff
@@ -1469,3 +1469,732 @@
+	var/turf/target_turf = get_turf(src)
+	if(!target_turf)
+		ScrapeAway()
+		return
```
That's impossible
The turf always exists, so get_turf(src) will always return src, and that if can never be hit
